cmake/modules/Findfmt.cmake

```diff
@@ -1,23 +1,23 @@
 find_path(fmt_INCLUDE_DIR NAMES fmt/format.h)
 
 if(fmt_INCLUDE_DIR)
-  set(_fmt_version_file "${fmt_INCLUDE_DIR}/fmt/format.h")
-  if(NOT EXISTS "${_fmt_version_file}")
-    set(_fmt_version_file "${fmt_INCLUDE_DIR}/fmt/core.h")
-  endif()
-  if(EXISTS "${_fmt_version_file}")
-    # parse "#define FMT_VERSION 40100" to 4.1.0
-    file(STRINGS "${fmt_INCLUDE_DIR}/fmt/format.h" fmt_VERSION_LINE
```

Comment:
@mogeb i think a simpler fix would just `s/${fmt_INCLUDE_DIR}/fmt/format.h/${_fmt_version_file}/`. less code churn, and less indention level. what do you think?

Reply:
The problem with that is that the current code assumes that the macro is defined in `format.h`, whereas it might be defined in `core.h`. Only `core.h` won't get searched if `format.h` exists, and the whole function fails.

In version 5, `FMT_VERSION` moved from `format.h` to `core.h`.

Reply:
@mogeb 

> The problem with that is that the current code assumes that the macro is defined in `format.h`, whereas it might be defined in `core.h`.

i agree. but not quite true, because it was an oversight. see

https://github.com/ceph/ceph/blob/dfc1c788894d992738dae3ba839a4cfb342f8cb1/cmake/modules/Findfmt.cmake#L4-L7

so, what i suggest is to replace the "${fmt_INCLUDE_DIR}/fmt/format.h" with "${_fmt_version_file}" in following line

https://github.com/ceph/ceph/blob/dfc1c788894d992738dae3ba839a4cfb342f8cb1/cmake/modules/Findfmt.cmake#L10

Reply:
But `format.h` still exists :) Even if I did the replacement you suggest, `_fmt_version_file` still resolves to `format.h`, which is the wrong file to search.

Actually the right thing to do here is to first assume it's in `core.h`, and only check `format.h` if `core.h` doesn't exist. So just invert the files here https://github.com/ceph/ceph/blob/dfc1c788894d992738dae3ba839a4cfb342f8cb1/cmake/modules/Findfmt.cmake#L4-L7 - I'll update the PR :)

Reply:
ahh, i see! so in libfmt 5.x, both files exist. but only `core.h` has the version string. but in 4.x, only `format.h` exists.